src/microbe_stage/ProcessSystem.cs

```diff
@@ -377,81 +381,115 @@ private void ProcessNode(IProcessable processor, float delta, float inverseDelta
         }
 
         if (environmentModifier <= MathUtils.EPSILON)
-        {
             canDoProcess = false;
-        }
 
+        // Compute spaceConstraintModifier before updating the final use and input amounts
         foreach (var entry in processData.Inputs)
         {
             if (entry.Key.IsEnvironmental)
                 continue;
 
-            var inputRemoved = entry.Value * process.Rate * delta;
-
-            currentProcessStatistics?.AddInputAmount(entry.Key, inputRemoved * inverseDelta);
+            var inputRemoved = entry.Value * process.Rate * delta * environmentModifier * spaceConstraintModifier;
 
             // currentProcessStatistics?.AddInputAmount(entry.Key, 0);
```

Comment:
```suggestion
```
pretty clearly looks like code from a past age.

Reply:
I kept this, for now, because the process panel has multiple issues open against it:
https://github.com/Revolutionary-Games/Thrive/issues/2085
https://github.com/Revolutionary-Games/Thrive/issues/1982
https://github.com/Revolutionary-Games/Thrive/issues/2295

Reply:
This feels like a weak argument to me, but if you really want to keep it I would suggest at least adding an explanatory comment. Future devs aren't going to guess that.

Reply:
I'm just keeping these lines, though. So I'm going to pull the "it was this way already" argument as to why these lines are here.

Reply:
fair enough